src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -633,6 +642,24 @@ public void deleteInfrastructure(Infrastructure infrastructure) {
         autoGeneratedSecurityGroups.remove(infrastructure.getId());
     }
 
+    private String getIdOrNameFromSecurityGroupName(Infrastructure infrastructure, SecurityGroupApi securityGroupApi,
+            Map.Entry<String, String> securityGroupInfos) {
+        if (securityGroupInfos.getValue() != null) {
+            return securityGroupInfos.getValue();
+        }
+        String securityGroupName = securityGroupInfos.getKey();
+        Set<SecurityGroup> sgs = securityGroupApi.describeSecurityGroupsInRegionWithFilter(infrastructure.getRegion(),
+                                                                                           ImmutableMultimap.of("group-name",
+                                                                                                                securityGroupName));
+        if (sgs.isEmpty()) {
```

Comment:
just in case, you should use `sgs == null || sgs.isEmpty()`

Reply:
you should maybe handle exceptions thrown by describeSecurityGroupsInRegionWithFilter

Reply:
In my tests, sgs was empty if no security groups has been found. But I can add the test anyway.
`describeSecurityGroupsInRegionWithFilter` does not throw any exception. But just in case, my PR on `nodesource-addons` handles a possible `RuntimeException` during the whole infrastructure termination call.